### PR TITLE
History service

### PR DESCRIPTION
**Pull request by kumsssss**
1. Create history service

### CODE REVIEW

File: history-service/controller/history-controller.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { ormGetUserQuestionHistory as _getUserQuestionHistory, ormAddQuestionToHistory as _addQuestionToHistory } from "../model/history-orm.js";
+
+export async function getUserQuestionHistory(req, res) {
+    try {
+        let userName = req.params.user;
+        if (!userName) {
+            return res.status(404).json({ message: "userName must be provided when requesting question history" });
+        }
+        const questions = await _getUserQuestionHistory(userName);
+        res.status(200).json({ questionHistory: questions });
```
Comment (lerxcl) [inline history-service/controller/history-controller.js:10]: I think we can standardize and just `return` for all the responses

File: history-service/controller/history-controller.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    } catch (err) {
+        res.status(500).json({message: "Database failure when trying to get questions"});
```
Comment (lerxcl) [inline history-service/controller/history-controller.js:12]: Same here

**Review by lerxcl — COMMENTED** [review]
I think you can also provide a README.md to specify what we need in our `.env` file, thanks 😄

**Review by lerxcl — APPROVED** [review]
LGTM now!

File: history-service/model/repository.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+
+export async function getQuestionHistory(userName) {
+    return HistoryModel.find({ userName: userName }).select('title topic difficulty -_id');
+}
+
+export async function addQuestionToHistory(data) {
```
Comment (Md-Fazil) [inline history-service/model/repository.js:19]: If a user attempts the same question multiple times, we will have multiple entries that will be returned to the frontend and appear like a log of the attempts. Is this intended?

File: history-service/model/repository.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+let mongoDB = process.env.ENV == "PROD" ? process.env.CLOUD_DB_URL : process.env.LOCAL_DB_URL;
+
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+
+export async function getQuestionHistory(userName) {
+    return HistoryModel.find({ userName: userName }).select('title topic difficulty -_id');
```
Comment (Md-Fazil) [inline history-service/model/repository.js:16]: If we only want to display distinct questions, we can add a distinct clause and order by the record creation time in descending order to allow users to view latest question solved first.

File: history-service/model/repository.js (hunk 3 above)
Comment (kumsssss) [reply history-service/model/repository.js:19]: now each question for a user will only be recorded once(last attempted) even if he does the same question multiple times.

File: history-service/model/repository.js (hunk 4 above)
Comment (kumsssss) [reply history-service/model/repository.js]: 👍

**Pull request merged**